Subject: [PATCH] ConnectivityService doesn't send decreased score to factories at need

When there is more than one network, after Score of a network fell down
requests of that network are removed from its "networkRequests" during
process of rematchNetworkAndRequests().

As a result, current implementation of updateNetworkScore()
cannot send score of that network in sendUpdatedScoreToFactories().
This may prevent other network factories from updating their status,
e.g. TelephonyNetworkFactory cannot enable APN.

Therefore, we add process of sending decreased score into
rematchNetworkAndRequests() to fix the issue.

diff --git a/services/core/java/com/android/server/ConnectivityService.java b/services/core/java/com/android/server/ConnectivityService.java
index 039b632b1b7..4da4fce7b81 100644
--- a/services/core/java/com/android/server/ConnectivityService.java
+++ b/services/core/java/com/android/server/ConnectivityService.java
@@ -5015,6 +5015,9 @@ private void rematchNetworkAndRequests(NetworkAgentInfo newNetwork,
                     if (VDBG) log("rematch for " + newNetwork.name());
                     if (currentNetwork != null) {
                         if (VDBG) log("   accepting network in place of " + currentNetwork.name());
+                        // Also need to send score of network which is being replaced
+                        sendUpdatedScoreToFactories(nri.request, currentNetwork.getCurrentScore());
+
                         currentNetwork.removeRequest(nri.request.requestId);
                         currentNetwork.lingerRequest(nri.request, now, mLingerDelayMs);
                         affectedNetworks.add(currentNetwork);
